h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeNode.java

```diff
@@ -443,6 +440,37 @@ void printDotEdges(PrintStream os, int maxLevelsToPrintPerEdge, float totalWeigh
     }
   }
 
+  public Map<String, Object> toJson() {
+    Map<String, Object> json = new HashMap<>();
+    json.put("nodeNumber", nodeNumber);
+    if (!Float.isNaN(weight)) json.put("weight", weight);
+    json.put("depth", depth);
+    json.put("colId", colId);
+    json.put("colName", colName);
+    json.put("leftward", leftward);
+    json.put("naVsRest", naVsRest);
+    json.put("inclusiveNa", inclusiveNa);
+    json.put("isCategorical", isBitset());
+    if (!Float.isNaN(predValue)) json.put("predValue", predValue);
+    if (!Float.isNaN(squaredError)) json.put("squaredError", squaredError);
+    if (domainValues != null && bs != null) {
+      List<String> matchedDomainValues = new ArrayList<>();
```

Comment:
just keep in mind that the domain can be huge and so can be the json files

Reply:
is there something we can do about that? use the bitset representation somehow?

Reply:
the easiest thing to do would be to refer to the values using numerical indices and keep the mapping in a separate file

decoding the bitset would also be possible

I think we do not have to overcomplicate it for now - just something to keep in mind for the future